hack/validate-rbac-resourcenames.sh

```
@@ -0,0 +1,156 @@
+extract_kubebuilder_resourcenames() {
+    local resource_type="$1"
+    grep -E "^\s*//\s*\+kubebuilder:rbac:.*resources=${resource_type}.*resourceNames=" \
+        "${PROJECT_ROOT}/pkg/controller/external_secrets/controller.go" | \
+        sed -E 's/.*resourceNames=([^,]*).*/\1/' | \
+        tr ';' '\n' | sort -u
+}
+
+# extract actual resource names from assets
+extract_asset_names() {
+    local pattern="$1"
+    find "${PROJECT_ROOT}/bindata/external-secrets/resources" "${PROJECT_ROOT}/bindata/external-secrets" -name "*.yml" -exec grep -l "kind: ${pattern}" {} \; 2>/dev/null | \
```

Shellcheck (0.10.0)

[warning] Use 'find .. -print0 | xargs -0 ..' or 'find .. -exec .. +' to allow non-alphanumeric filenames.

(SC2038)